src/Sarif.Viewer.VisualStudio/SarifError.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+using Microsoft.VisualStudio.Text;
+
+namespace SarifViewer
+{
+    public class SarifError
+    {
+        public SarifError(string fileName)
+        {
+            FileName = fileName;
+        }
+
+        public string Tool { get; set; }
+
+        public string MimeType { get; set; }
+
+        public bool RegionPopulated { get; set; }
+
+        public Region Region { get; set; }
+
+        public string FileName { get; set; }
+
+        public string Message { get; set; }
+
+        public SnapshotSpan Span { get; set; }
+
+        public int LineNumber { get; set; }
+
+        public int ColumnNumber { get; set; }
+
+        public bool IsError { get; set; } 
+
+        public string ErrorCode { get; set; }
+         
+        public string HelpLink { get; set; }
+
+        public override string ToString()
+        {
+            return Message;
+        }
```

Comment:
it's not clear yet that this object needs to be immutable. we have the RegionPopulated bool, for example, that tells us whether we've expanded all region data (this is a lazy operation because it requires reading target source file and caching a newline index).

we'll keep an eye on this situation, SarifError is not a finished construct.